app/models/user.rb

```diff
@@ -9,4 +9,12 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friend_requests
+  has_many :friends, class_name: 'FriendRequest', foreign_key: 'friend_id'
```

Comment:
Like the previous reviewer commented, you need to make a small change here. 
The friends are only friends when they accept the invitation. You should use a `where` to find the `confirmed: true` cases.
You are using a non-existing foreign_key. If you try your association in the console, it will never find a friend for a user because you don't have a column called `friend_id` in your `FriendRequest` table.

Reply:
We have fixed the ` has_many :friends, class_name: 'FriendRequest', foreign_key: 'friend_id'` to return only the records where `confirmed: true`.
However, the foreign key `friend_id` is not non-existent as we are using it in our `FriendRequest` Table.